jwst/ramp_fitting/ramp_fit.py

```diff
@@ -242,26 +582,32 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
     #   is the number of given by the NFRAMES keyword, and is the number of
     #   frames averaged on-board for a group, i.e., it does not include the
     #   groupgap.
-    effintim, nframes, groupgap, dropframes1 = utils.get_efftim_ped(model)
+
+    effintim = (nframes + groupgap) * frame_time
 
     # Get GROUP DQ and ERR arrays from input file
-    gdq_cube = model.groupdq
+    gdq_cube = groupdq
     gdq_cube_shape = gdq_cube.shape
 
     # If all the pixels have their initial groups flagged as saturated, the DQ
     #   in the primary and integration-specific output products are updated,
     #   the other arrays in all output products are populated with zeros, and
     #   the output products are returned to ramp_fit(). If the initial group of
     #   a ramp is saturated, it is assumed that all groups are saturated.
-    first_gdq = model.groupdq[:,0,:,:]
+    first_gdq = groupdq[:,0,:,:]
     if np.all(np.bitwise_and( first_gdq, dqflags.group['SATURATED'] )):
-        new_model, int_model, opt_model = utils.do_all_sat( model, imshape,
-                                                            n_int, save_opt )
+        new_model, int_model, opt_model = utils.do_all_sat(inpixeldq, groupdq, imshape, n_int, save_opt )
+        if (save_opt):
+            actual_segments = 0
+            actual_CRs = 0
 
-        return new_model, int_model, opt_model
+        return new_model.data, new_model.dq, new_model.var_poisson, new_model.var_rnoise, new_model.err, \
```

Comment:
These 20+ values returned could be consolidated, if the 3 separate models were returned.

Reply:
I agree that returned this many values is awkward but the models cannot be returned directly. This is because the multiprocessing process uses pickle to pass the parameters to the parallel process and the data models cannot be pickled due to something odd in the ASDF format.
I spent a while trying alternative methods but the only thing that works is passing the numpy arrays in and out as parameters.

Reply:
Hmmm. I will let you know if I have any ideas.